users/views.py

```diff
@@ -90,13 +94,23 @@ def post(self, request, *args, **kwargs):
         return Response(serializer.errors, status=HTTP_400_BAD_REQUEST)
 
 
-class UserDetail(mixins.RetrieveModelMixin, generics.GenericAPIView):
+class UserDetail(generics.RetrieveUpdateAPIView):
     queryset = User.objects.all()
-    serializer_class = UserSerializer
+    serializer_class = UserDetailSerializer
 
     def get(self, request, *args, **kwargs):
         return self.retrieve(request, *args, **kwargs)
 
+    @method_is_authenticated
+    @method_user_is_self
+    def put(self, request, *args, **kwargs):
+        return self.update(request, *args, **kwargs)
+
+    @method_is_authenticated
```

Comment:
What will be the different use cases for each of these?

Reply:
The RetrieveUpdateAPIView implements both by default, so I'm overriding both as well in order to include the permissions decorators. I figure it gives us the flexibility of using either method in whatever way feels fitting on the frontend.